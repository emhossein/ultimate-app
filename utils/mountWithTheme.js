import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@emotion/react';
import theme from '../config/theme';

const mountWithTheme = ({ children }) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
};

export default mountWithTheme;
