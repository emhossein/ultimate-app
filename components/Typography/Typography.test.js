/**
 * @jest-environment jsdom
 */

import React from 'react';
import { createSerializer } from '@emotion/jest';
import mountWithTheme from '../../utils/mountWithTheme';
import { H1, H2, H3, Paragraph } from '.';

expect.addSnapshotSerializer(createSerializer());

describe('Typography', () => {
  test('H1 Renders Correctly', () => {
    const component = mountWithTheme(<H1>Heading 1</H1>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('H2 Renders Correctly', () => {
    const component = mountWithTheme(<H2>Heading 2</H2>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('H3 Renders Correctly', () => {
    const component = mountWithTheme(<H3>Heading 3</H3>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Paragraph Renders Correctly', () => {
    const component = mountWithTheme(<Paragraph>Paragraph</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
