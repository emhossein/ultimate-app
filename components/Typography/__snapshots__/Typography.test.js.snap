// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography H1 Renders Correctly 1`] = `null`;

exports[`Typography H2 Renders Correctly 1`] = `null`;

exports[`Typography H3 Renders Correctly 1`] = `null`;

exports[`Typography Paragraph Renders Correctly 1`] = `null`;
